audio/analyzer.py

```diff
@@ -111,21 +112,67 @@ def analyze_album(album_path, upload_spectrogram=False):
     flac_files = [file for file in os.listdir(album_path) if file.lower().endswith('.flac')]
     album = {}
 
+    # Ask the user for input if upload_spectrogram is True
+    tracks_to_upload_spectrograms = set()
+    if upload_spectrogram:
+        print("Enter the track numbers to upload spectrals separated by commas, or '*' for all tracks:")
+        user_input = input().strip()
+        if user_input == '*':
+            # Use a wildcard to indicate all tracks for spectrogram upload
+            tracks_to_upload_spectrograms = {'*'}
+        else:
+            # Parses user input, keeping the disc prefix and removing leading zeros from track numbers
+            tracks_to_upload_spectrograms = set(track.strip().lstrip('0') for track in user_input.split(','))
+
     with alive_bar(len(flac_files), title='Processing tracks', bar='classic', spinner='classic', length=40) as bar:
         for file in flac_files:
             file_path = os.path.join(album_path, file)
-            flac_info = get_flac_info(file_path, upload_spectrogram)
+            track_number = get_track_number(file_path)  # This function should return the track number as a string
+            # Determine if this track's spectrogram should be uploaded
+            should_upload_spectrogram = ('*' in tracks_to_upload_spectrograms or
+                                         any(track_number == track.lstrip('0') for track in tracks_to_upload_spectrograms))
+            flac_info = get_flac_info(file_path, should_upload_spectrogram)
             file_name = os.path.basename(file_path)
             album[file_name] = flac_info
             bar()
 
     try:
-        album = dict(sorted(album.items(), key=lambda x: int(x[1]['#']), reverse=False))
+        album = dict(sorted(album.items(), key=lambda item: (int(item[1]['disc_number']), int(item[1]['#'])), reverse=False))
     except:
         pass
 
     return album
 
+# Now this considers how I format tracks with Musicbee. Hopefully I've made it work with other tagging types
+def get_track_number(file_path):
```

Comment:
This function is more or less redundant, since this also gets saved in the Track metadata. Using the metadata is more robust since it doesn't relay on Naming conventions